src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2199,7 +2211,7 @@ private void doEncodeConnack(
             int propertiesSize = 0;
 
             MqttPropertyFW mqttProperty;
-            if (sessionExpiryIntervalLimit > 0)
+            if (sessionExpiryInterval > sessionExpiryIntervalLimit && sessionExpiryIntervalLimit > 0)
```

Comment:
What if we made the default limit `unlimited`, i.e. `Integer.MAX_VALUE`, then this logic could revert to the simpler form of `if (sessionExpiryInterval > sessionExpiryIntervalLimit)`, agree?